/** (C) Copyright 2013, Applied Physical Sciences Corp., A General Dynamics Company
 **
 ** Gravity is free software; you can redistribute it and/or modify
 ** it under the terms of the GNU Lesser General Public License as published by
 ** the Free Software Foundation; either version 3 of the License, or
 ** (at your option) any later version.
 **
 ** This program is distributed in the hope that it will be useful,
 ** but WITHOUT ANY WARRANTY; without even the implied warranty of
 ** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 ** GNU Lesser General Public License for more details.
 **
 ** You should have received a copy of the GNU Lesser General Public
 ** License along with this program;
 ** If not, see <http://www.gnu.org/licenses/>.
 **
 */

option optimize_for = SPEED;
option java_outer_classname = "GravityDataProductContainer";
option java_package = "com.aphysci.gravity.protobuf";

package gravity;

message GravityDataProductPB
{
	optional uint64 timestamp = 1;
	optional string dataProductID = 2;
	optional string softwareVersion = 3;
	optional bytes data = 4;
	optional string componentID = 5;
	optional string domain = 6;
	optional bool future_response = 7;
	optional string future_socket_url = 8;
	optional uint64 received_timestamp = 9;
	optional bool is_cached_dataproduct = 10;
	optional bool is_relayed_dataproduct = 11 [default = false];
	optional string protocol = 12 [default="protobuf2"];
	optional string type_name = 13 [default=""]; // e.g.: gravity.GravityDataProductPB
}

